mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,394 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
+	"github.com/hashicorp/terraform-plugin-mux/tf6muxserver"
+	sdkv2schema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	cstmvalidator "github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/framework/validator"
+)
+
+type MongodbtlasProvider struct{}
+
+type tfMongodbAtlasProviderModel struct { //nolint:govet //prefer to maintain order of attributes over small byte optimization
+	PublicKey            types.String `tfsdk:"public_key"`
+	PrivateKey           types.String `tfsdk:"private_key"`
+	BaseURL              types.String `tfsdk:"base_url"`
+	RealmBaseURL         types.String `tfsdk:"realm_base_url"`
+	IsMongodbGovCloud    types.Bool   `tfsdk:"is_mongodbgov_cloud"`
+	AssumeRole           types.List   `tfsdk:"assume_role"`
+	SecretName           types.String `tfsdk:"secret_name"`
+	Region               types.String `tfsdk:"region"`
+	StsEndpoint          types.String `tfsdk:"sts_endpoint"`
+	AwsAccessKeyID       types.String `tfsdk:"aws_access_key_id"`
+	AwsSecretAccessKeyID types.String `tfsdk:"aws_secret_access_key"`
+	AwsSessionToken      types.String `tfsdk:"aws_session_token"`
+}
+
+type tfAssumeRoleModel struct { //nolint:govet //prefer to maintain order of attributes over small byte optimization
+	Duration          types.String `tfsdk:"duration"`
+	DurationSeconds   types.Int64  `tfsdk:"duration_seconds"`
+	ExternalID        types.String `tfsdk:"external_id"`
+	Policy            types.String `tfsdk:"policy"`
+	PolicyARNs        types.Set    `tfsdk:"policy_arns"`
+	RoleARN           types.String `tfsdk:"role_arn"`
+	SessionName       types.String `tfsdk:"session_name"`
+	SourceIdentity    types.String `tfsdk:"source_identity"`
+	Tags              types.Map    `tfsdk:"tags"`
+	TransitiveTagKeys types.Set    `tfsdk:"transitive_tag_keys"`
+}
+
+func (p *MongodbtlasProvider) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = "mongodbatlas"
+	// resp.Version = "" we can look into defining this value automatically for release process
```

Comment:
is this a left over? if not, could you add addition info about when this code will be removed? A jira ticket is enough

Reply:
yes, was left over. The Version attribute is currently not being used for any functionality (https://github.com/hashicorp/terraform-plugin-framework/blob/55f0d6bc4be2a57adb06a5c25c28640712c88129/provider/metadata.go#L21C2-L21C2).
Looking more into this found the version is easily accessible, so will include now.